### PR TITLE
Add "Padding" slider to control distance from edge while docked

### LINKED ISSUE #55: Specify constant distance from edge
When docked to the bottom of the screen, Touch Bar Simulator covers up dock icon labels. When I used the floating window option before docking was a thing, I would always float the window 20px or so above the dock to make room for the dock labels.

Would be great to be able to define this, possibly with a lil slider in the menu bar above "transparency".

If this feature is added, my recommendation is to have all four corners be rounded if the Touch Bar simulator window is not directly against the edge.

### PR DESCRIPTION
**Pull request by ThatsJustCheesy**
Closes #55.

Also upgrades Defaults.framework to v3.1.1 and uses its tieToLifetime(of:), obsoleting (and removing) the version in glue.swift.

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -1,3 +1,3 @@
-github "sindresorhus/Defaults" "v3.0.0"
+github "sindresorhus/Defaults" "v3.1.1"
```
Comment (sindresorhus) [inline Cartfile.resolved]: Can you move this to be a Swift Package Manager dependency in Xcode instead?

**Comment by sindresorhus** [comment]
I think the "Padding" slider should be disabled if the mode is "Floating".

**Comment by ThatsJustCheesy** [comment]
Agreed. I had considered that, but simply never got to doing it.On Nov 24, 2019 5:06 AM, Sindre Sorhus <notifications@github.com> wrote:I think the "Padding" slider should be disabled if the mode is "Floating".

**Comment by sindresorhus** [comment]
@ThatsJustCheesy Bump :)

**Comment by ThatsJustCheesy** [comment]
Apologies for the wait, but I think it's done now :)

**Comment by sindresorhus** [comment]
No worries. Really nice work on this, by the way. 👌

**Pull request merged**